stdlib/InteractiveUtils/src/editless.jl

```diff
@@ -5,6 +5,169 @@
+
+Define a new editor matching `pattern` that can be used to open a file
+(possibly at a given line number) using `fn`.
+
+The `fn` argument is a function that determines how to open a file with the
+given editor.  It should take 2 or 3 arguments, as follows:
+
+* `command` - an array of strings representing the editor command. It can be
+  safely interpolated into a command created using backtick notation.
+* `path`  - the path to the source file to open
+* `line` - the optional line number to open to; if specified the returned command
+  must open the file at the given line.
+
+`fn` must return either an appropriate `Cmd` object to open a
+file, a function (taking 0 arguments) that will open the file directly
+(returning a `Cmd` is the preferred approach), or `nothing`. Use `nothing`
+to indicate that this editor is not appropriate for the current environment
+and another editor should be attempted.
+
+The `pattern` argument is a string, regular expression, or an array of strings
+and regular expressions. For the `fn` to be called one of the patterns must
+match the value of `EDITOR`, `VISUAL` or `JULIA_EDITOR`.  For strings, only
+whole words can match (i.e. "vi" doesn't match "vim -g" but will match
+"/usr/bin/vi -m").
+
+If multiple defined editors match, the one with the highest priority is
+selected, and if there is a tie in priority, the first editor added will be
+used.
+
+By default julia does not wait for the editor to close, running it in the
+background. However, if the editor is terminal based, you will probably want to
+set `wait=true` and julia will wait for the editor to close before resuming.
+
+If no editor entry can be found, then a file is opened by running
+`\$command \$path`.
+
+Note that a number of default editors (all priority 0) are already defined. All
+of the following commands should already work:
+
+- emacs
+- vim
+- nvim
+- nano
+- textmate
+- mate
+- kate
+- subl
+- atom
+- notepad++
+- Visual Studio Code
+- open
+
+# Example:
+The following defines the usage of terminal-based `emacs`:
+
+    define_editor(r"\bemacs\b.*(-nw|--no-window-system)",
+                  wait=true) do cmd, path, line
+        `\$cmd +\$line \$path`
+    end
+"""
+
+function define_editor(fn, pattern; wait=false, priority=0)
+    nargs = map(x -> x.nargs - 1, methods(fn).ms)
+    has3args = 3 ∈ nargs
+    has2args = 2 ∈ nargs
+    entry = EditorEntry(pattern, fn, wait, has3args, Float64(priority))
+    push!(editors, entry)
+
+    if !(has3args || has2args)
+        error("Editor function must take 2 or 3 arguments")
+    end
+end
+
+function define_default_editors()
+    define_editor(["vim","vi","nvim","mvim","nano"],
+                  wait=true) do cmd, path, line
+        `$cmd +$line $path`
+    end
+    define_editor(r"\bemacs\b.*(-nw|--no-window-system)",
+                  wait=true) do cmd, path, line
+        `$cmd +$line $path`
+    end
+    define_editor(r"\bemacsclient\b.*(-nw|-t|-tty)",
```

Comment:
Note that this allows `--nw` and `--tty` and (accidentally?) solves the issue I reported in #27410.

Reply:
So it does! I wish I could say that was intentional.

Reply:
```suggestion
    define_editor(r"\bemacsclient\b.*\s-(t|-?nw|-?tty)\b",
```